### PR TITLE
updated to version 8 and LDV2 compatibility

### LINKED ISSUE #34: update locked-deal version to 2
https://github.com/The-Poolz/synthetic-token/blob/020274de60cb96379b69a5f013db5b14234366c9/contracts/Token.sol#L74

### LINKED ISSUE #35: update solidity version to 8
https://github.com/The-Poolz/synthetic-token/blob/020274de60cb96379b69a5f013db5b14234366c9/contracts/Token.sol#L3

### PR DESCRIPTION
**Pull request by YouStillAlive**

### CODE REVIEW

File: contracts/Manageable.sol [hunk 1]
```diff
@@ -1,74 +1,82 @@
...
 
     function _SetLockingDetails(
         address _tokenAddress,
         uint256 _amount,
-        uint64[] memory _unlockTimes,
-        uint8[] memory _ratios,
-        uint256 _finishTime
+        uint64 _startLockTime,
+        uint64 _finishLockTime,
+        uint256 _endTime
```
Comment (Lomet) [inline contracts/Manageable.sol]: let's keep the ratios and the arrays.
we can build a nice shaping on the charts

File: contracts/OriginalToken.sol [hunk 2]
```diff
@@ -13,8 +13,7 @@ contract OriginalToken is ERC20, Ownable{
     * @dev assign totalSupply to account creating this contract
     */
     
-    constructor(string memory _TokenName, string memory _TokenSymbol) ERC20(_TokenName, _TokenSymbol) public {
-        _setupDecimals(18);
+    constructor(string memory _TokenName, string memory _TokenSymbol) ERC20(_TokenName, _TokenSymbol) {
```
Comment (Lomet) [inline contracts/OriginalToken.sol]: let's make this part pretty

File: contracts/ERC20WithDecimals.sol [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    ) internal virtual override {
+        require(
+            EndTime <= block.timestamp ||
+                _msgSender() == owner() ||
+                to == address(0) ||
+                registerWhitelist(to, amount),
+            "Invalid Transfer Time or To Address"
+        );
+        super._beforeTokenTransfer(from, to, amount); // Call parent hook
+    }
```
Comment (Lomet) [inline contracts/ERC20WithDecimals.sol]: this part not fit the name

File: contracts/Manageable.sol [hunk 4]
```diff
@@ -1,74 +1,82 @@
...
-    uint256 public FinishTime;
+    uint256 public EndTime;
+
+    bool public LockStatus;
 
     struct lockDetails {
-        uint64 unlockTime;
-        uint ratio;
+        uint64 startTime;
+        uint64 finishTime;
```
Comment (Lomet) [inline contracts/Manageable.sol:29]: with ratios

File: contracts/Token.sol [hunk 5]
```diff
@@ -1,122 +1,112 @@
...
+        _SetLockingDetails(
+            _tokenAddress,
+            cap(),
+            _startLockTime,
+            _finishLockTime,
+            _endTime
+        );
     }
 
     function ActivateSynthetic() external {
```
Comment (Lomet) [inline contracts/Token.sol]: let's call this withdrawal, the same as in Lockdeal

File: package.json [hunk 6]
```diff
@@ -12,7 +13,7 @@
     "ganache-time-traveler": "^1.0.15",
     "moonbeam-truffle-plugin": "^1.1.6",
     "node-fetch": ">=2.6.1",
-    "poolz-helper": "^1.1.6",
+    "poolz-helper-v2": "^2.0.16",
```
Comment (Lomet) [inline package.json]: 👍

**Review by Lomet — COMMENTED** [review]
Good work, keep it up.
need to make this with ratios

**Review by ashwinarora — APPROVED** [review]

**Review by Lomet — APPROVED** [review]

**Pull request merged**